### DISCUSSION TITLE
More H1 registers - testers needed

### DISCUSSION THREAD

**Discussion by canton7** [nathanmarlor/foxess_modbus#65] [General]
Hi,

I've identified the following new registers for the H1 over RS485:

* 11070: PV total * 10
* 11071: PV today * 10
... Presumably some in the range 11072-11078 are for charge / discharge, but I'm still waiting on my battery so can't test
* 11079: Feed-in total * 10
* 11080: Feed-in today * 10
* 11082: Consume total * 10  (as shown under History on the inverter. Is this grid consumption?)
* 11083: Consume today * 10 (as shown under History on the inverter)
... Presumably EPS is next, but I can't test
* 11085: I think this is 'Energy Generated Total' * 10. The number's about right, but I can't find anything to compare it to
* 11086: I think this is 'Energy Generated Today' * 10

If anyone has the chance, could you please:

1. Check that what I've written above is correct for you. Bonus points if you have an inverter other than the H1, so we can make these available for more models
2. Look at the range 11072-11078, see if you can spot charge/discharge stats
3. Look for EPS stats

Thanks!

**Comment by nathanmarlor** [comment]
@canton7 I had a very quick run through of these earlier but couldn't correlate against Fox Cloud - I'll have another look tomorrow 👍

**Reply by canton7** [reply]
Cheers! Fox cloud seemed a little bit off on some of these for me as well, interestingly. I matched them against the inverter directly, under History -> Inverter Yield on the front panel.

**Reply by nathanmarlor** [reply]
Good shout - I'll have a look in the AM. Will leave the PR pending until then if that's okay?

**Reply by canton7** [reply]
No problem, thank you!

**Comment by nathanmarlor** [comment]
Just checking you meant * 0.1 and not * 10?

11070 - agreed, PV total yield
11071 - agreed, PV yield today

11073 - charge total
11074 - charge today

11076 - discharge total
11077 - discharge today

11079 - agreed, feed in total
11080 - agreed, feed in today

11082 - agreed, consume total
11083 - agreed, consume today

EPS doesn't seem to appear. I'll get these integrated now.

**Reply by canton7** [reply]
Perfect, thank you!

By *10 I meant that the value read over modbus is 10x the value in kW/kWh. So we need to apply a scaling factor of 0.1 to turn the value into kW/kWh. I think we're on the same page, just me being unclear.

**Reply by canton7** [reply]
I wonder what appears in those missing registers -- 11072, 11075, etc. There's a fairly obvious grouping of 3 there, and some of them have a value when I read them.

**Reply by nathanmarlor** [reply]
They were all 0s for me. Bizzare that EPS didn't make an appearance too.

**Reply by canton7** [reply]
11072: 0
11075: 0
11078: 110
11081: 58
11084: 114

**Reply by nathanmarlor** [reply]
Great work on this BTW - it now rips even more yaml out of the HA configuration as battery/feed-in/consume total are now provided natively. :rocket:

**Reply by nathanmarlor** [reply]
> 11072: 0 11075: 0 11078: 110 11081: 58 11084: 114

![image](https://user-images.githubusercontent.com/3436911/227475460-0f7a590c-87b4-4856-b834-1ae20e5e62e6.png)

**Reply by canton7** [reply]
Oh FFS, it's this stupid W610 again. Turns out that on modpoll on Windows, if you try and read more than 8 registers, you just get garbage. The linux version at least shows 0's.

So no, all of those other registers *are* reading 0 for me.

**Reply by canton7** [reply]
> Great work on this BTW - it now rips even more yaml out of the HA configuration as battery/feed-in/consume total are now provided natively. 🚀

Thanks! I wanted a "PV Today" sensor without needing to integrate, and noticed that the inverter and foxcloud knew this value, so went hunting for registers with the right value...

**Reply by nathanmarlor** [reply]
Do you have a single string? As still need to integrate PV1/PV2 in order to get:

![image](https://user-images.githubusercontent.com/3436911/227476788-004c8e9d-60ba-4f20-b70a-34be0f267594.png)

**Reply by canton7** [reply]
Yeah, I've only got limited roof space, so just a single string of 8 panels for me.
